### DISCUSSION TITLE
Help about wiring

### DISCUSSION THREAD

**Discussion by bilalhp** [alienatedsec/solis-ha-modbus-cloud#10] [Q&A] [answered]
Hello,

I have some questions about wiring:

1. Should I connect both PGND and GND on the waveshare's to the same earth cable from outside?
2. Should I solder the Exceedconn side of the cables? Is there any alternatives?
3. The picture is a bit misleading about twisted pairs to use. Is it one wire per connection (A and B) or one pair for connection?

**Comment by alienatedsec** [comment, marked as answer]
> * Should I connect both PGND and GND on the waveshare's to the same earth cable from outside?

I would not connect to the same cable.

> * Should I solder the Exceedconn side of the cables? Is there any alternatives?

I haven't seen similar connectors that don't require soldering.

> * The picture is a bit misleading about twisted pairs to use. Is it one wire per connection (A and B) or one pair for connection?

As mentioned in [this post](https://github.com/wills106/homeassistant-solax-modbus/discussions/188#discussioncomment-5729312), there are two wires twisted together - e.g. orange and white-orange - you connect orange to RS485(A) and white-orange to RS485(B). Why twisted? I cannot explain better than the below source:

https://www.analog.com/en/technical-articles/rs485-cable-specification-guide--maxim-integrated.html

**Reply by bilalhp** [reply]
`I would not connect to the same cable.`

How did you connect them in your case?

**Reply by alienatedsec** [reply]
I don't have ground connected - just earth cabling. The reason is that I don't expect it to overload with PoE. However, electric shocks are likely to occur, and it's better to be released somewhere if those are to happen.
